pandas/core/indexes/multi.py

```diff
@@ -2156,6 +2156,8 @@ def _drop_from_level(self, codes, level, errors="raise"):
         i = self._get_level_number(level)
         index = self.levels[i]
         values = index.get_indexer(codes)
+        nan_codes = isna(codes)
+        values[(np.equal(nan_codes, False)) & (values == -1)] = -2
```

Comment:
can you give a comment on this (similiar to your reply to me), also why seetting to -2 makes this work. ping on green.

Reply:
Done, will ping when green